src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -10464,6 +10464,40 @@ .locals init (System.Runtime.CompilerServices.DefaultInterpolatedStringHandler V
             VerifyMergedProperties(actualProperties, actualFields);
         }
 
+        [WorkItem("https://github.com/dotnet/roslyn/issues/75893")]
+        [Theory]
+        [CombinatorialData]
+        public void SpeculativeSemanticModel(bool includeLocal)
```

Comment:
Just FYI, doesn't have to be done in this PR, but when I previously worked on the feature, I have written relatively large amount of tests which also include SpeculativeSemanticModel. I haven't checked if the exact case in this test is one of those I wrote, but generally I think it may be a good idea to copy those tests. Some test outcomes may be different from the now-expected behavior and may require adjustments though.

The tests are here: https://github.com/dotnet/roslyn/blob/32d6c1db46fd613888d08970908257fbacf328ba/src/Compilers/CSharp/Test/Emit2/Semantics/PropertyFieldKeywordTests.cs

There was also this test:

https://github.com/dotnet/roslyn/blob/32d6c1db46fd613888d08970908257fbacf328ba/src/Compilers/CSharp/Test/Emit/PDB/PDBWinMdExpTests.cs#L246

And this:

https://github.com/dotnet/roslyn/blob/32d6c1db46fd613888d08970908257fbacf328ba/src/Compilers/CSharp/Test/Semantic/Semantics/InteractiveSemanticModelTests.cs#L99C21-L99C37

And https://github.com/dotnet/roslyn/blob/32d6c1db46fd613888d08970908257fbacf328ba/src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelAPITests.cs#L2011, and finally https://github.com/dotnet/roslyn/blob/32d6c1db46fd613888d08970908257fbacf328ba/src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTests.cs#L6959C21-L6959C52

Reply:
Thanks @Youssef1313 for links to your tests from the earlier branch. I had previously referenced the tests in `PropertyFieldKeywordTests` in https://github.com/dotnet/roslyn/issues/75398, and I've added a link to your comment there. I'll investigate the tests as part of that issue. In the meantime, this particular issue seemed straightforward enough to handle directly.